pkg/networkservice/loopback/server.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loopback
+
+import (
+	"context"
+
+	"git.fd.io/govpp.git/api"
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+type loopback struct {
+	vppConn api.Connection
+
+	loopbacks loopMap
+}
+
+// NewServer creates a NetworkServiceServer chain element to create the ip table in vpp
+func NewServer(vppConn api.Connection) networkservice.NetworkServiceServer {
+	return &loopback{
+		vppConn: vppConn,
+		loopbacks: loopMap{
+			entries: make(map[string]*loopInfo),
+		},
+	}
+}
+
+func (l *loopback) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	if err := createLoopback(ctx, request.GetConnection(), l.vppConn, &l.loopbacks, metadata.IsClient(l)); err != nil {
+		return nil, err
+	}
+	return next.Server(ctx).Request(ctx, request)
```

Comment:
There will be no timeout or Close if it is an initial Request and it fails further in the chain. Please add some cleanup for such case.

Reply:
@artem-belov Could you respond to this comment?

Reply:
Fixed